### PR TITLE
Remove manual byte editing in wallet_tx_clone func test

### PR DESCRIPTION
**Pull request by instagibbs**
Adapted from @stevenroose

### CODE REVIEW

**Comment by stevenroose** [comment]
Related: https://github.com/bitcoin/bitcoin/pull/15282/files

File: test/functional/wallet_txn_clone.py [hunk 1]
```diff
@@ -72,19 +74,14 @@ def run_test(self):
...
-            clone_raw = clone_raw[:pos0] + output1 + output0 + clone_raw[pos0 + 2 * output_len:]
+        clone_tx = CTransaction()
+        clone_tx.deserialize(io.BytesIO(bytes.fromhex(clone_raw)))
+        if (rawtx1["vout"][0]["value"] == 40 and clone_tx.vout[0].nValue != 40*COIN or rawtx1["vout"][0]["value"] != 40 and clone_tx.vout[0].nValue == 40*COIN):
+            (clone_tx.vout[0], clone_tx.vout[1]) = (clone_tx.vout[1], clone_tx.vout[0])
 
         # Use a different signature hash type to sign.  This creates an equivalent but malleated clone.
         # Don't send the clone anywhere yet
-        tx1_clone = self.nodes[0].signrawtransactionwithwallet(clone_raw, None, "ALL|ANYONECANPAY")
+        tx1_clone = self.nodes[0].signrawtransactionwithwallet(clone_tx.serialize().hex(), None, "ALL|ANYONECANPAY")
```
Comment (MarcoFalke) [inline test/functional/wallet_txn_clone.py]: Assigned 0.19.0, because this is python3.5 only

File: test/functional/wallet_txn_clone.py [hunk 2]
```diff
@@ -72,19 +74,14 @@ def run_test(self):
...
-        hex40 = "00286bee00000000"
-        output_len = 16 + 2 + 2 * int("0x" + clone_raw[pos0 + 16:pos0 + 16 + 2], 0)
-        if (rawtx1["vout"][0]["value"] == 40 and clone_raw[pos0:pos0 + 16] != hex40 or rawtx1["vout"][0]["value"] != 40 and clone_raw[pos0:pos0 + 16] == hex40):
-            output0 = clone_raw[pos0:pos0 + output_len]
-            output1 = clone_raw[pos0 + output_len:pos0 + 2 * output_len]
-            clone_raw = clone_raw[:pos0] + output1 + output0 + clone_raw[pos0 + 2 * output_len:]
+        clone_tx = CTransaction()
+        clone_tx.deserialize(io.BytesIO(bytes.fromhex(clone_raw)))
+        if (rawtx1["vout"][0]["value"] == 40 and clone_tx.vout[0].nValue != 40*COIN or rawtx1["vout"][0]["value"] != 40 and clone_tx.vout[0].nValue == 40*COIN):
+            (clone_tx.vout[0], clone_tx.vout[1]) = (clone_tx.vout[1], clone_tx.vout[0])
```
Comment (MarcoFalke) [inline test/functional/wallet_txn_clone.py]: nice swap

**Review by maflcko — APPROVED** [review]

File: test/functional/wallet_txn_clone.py (hunk 1 above)
Comment (instagibbs) [reply test/functional/wallet_txn_clone.py]: SGTM, I like the syntax way more. So should I ping once 0.18 is branched?

Comment (MarcoFalke) [reply test/functional/wallet_txn_clone.py]: Remind me #14954

File: test/functional/wallet_txn_clone.py (hunk 2 above)
Comment (practicalswift) [inline test/functional/wallet_txn_clone.py]: Nit: Parantheses redundant :-)

File: test/functional/wallet_txn_clone.py [hunk 3]
```diff
@@ -72,19 +74,14 @@ def run_test(self):
...
-        pos0 = 2 * (4 + 1 + 36 + 1 + 4 + 1)
-        hex40 = "00286bee00000000"
-        output_len = 16 + 2 + 2 * int("0x" + clone_raw[pos0 + 16:pos0 + 16 + 2], 0)
-        if (rawtx1["vout"][0]["value"] == 40 and clone_raw[pos0:pos0 + 16] != hex40 or rawtx1["vout"][0]["value"] != 40 and clone_raw[pos0:pos0 + 16] == hex40):
-            output0 = clone_raw[pos0:pos0 + output_len]
-            output1 = clone_raw[pos0 + output_len:pos0 + 2 * output_len]
-            clone_raw = clone_raw[:pos0] + output1 + output0 + clone_raw[pos0 + 2 * output_len:]
+        clone_tx = CTransaction()
+        clone_tx.deserialize(io.BytesIO(bytes.fromhex(clone_raw)))
+        if (rawtx1["vout"][0]["value"] == 40 and clone_tx.vout[0].nValue != 40*COIN or rawtx1["vout"][0]["value"] != 40 and clone_tx.vout[0].nValue == 40*COIN):
```
Comment (practicalswift) [inline test/functional/wallet_txn_clone.py]: Nit: Parantheses redundant :-)

**Comment by jonasschnelli** [comment]
Oh yes. Thanks for doing this.
utACK 6aaa0abc1241ad26f5895760701e77c9c8e93924 once 0.18 is branched off.

File: test/functional/wallet_txn_clone.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/wallet_txn_clone.py]: Already broken in `master`, so just going to merge this as well :woman_shrugging: 

```sh
$ git grep -l '\.hex()' test/
test/functional/mining_basic.py
test/functional/test_framework/wallet_util.py
test/functional/wallet_txn_clone.py

Comment (instagibbs) [reply test/functional/wallet_txn_clone.py]: Yeah at least partially my bad :1st_place_medal:

**Pull request merged**